.infra/ch_sync.ts

```diff
@@ -0,0 +1,287 @@
+  resourcePrefix?: string;
+  provider?: ProviderResource;
+  isAdhocEnv?: boolean;
+  disableHealthCheck?: boolean;
+};
+
+
+function deployClickhouseSyncSharedDependencies(
+  name: string,
+  {
+    resourcePrefix = '',
+    isAdhocEnv,
+  }: Pick<
+    OptionalArgs,
+    'resourcePrefix' | 'isAdhocEnv'
+  > = {},
+): {
+  chsyncSa: gcp.serviceaccount.Account | undefined;
+  chsyncKey: gcp.serviceaccount.Key | undefined;
+} {
+  if (isAdhocEnv) {
+    return {chsyncKey: undefined, chsyncSa: undefined};
+  }
+
+  // i don't think we need it as we don't use pubsub
+  const {serviceAccount: chsyncSa} = createServiceAccountAndGrantRoles(
+    `${resourcePrefix}chsync-sa`,
+    `${name}-chsync`,
+    `${name}-chsync`,
+    [],
+    isAdhocEnv,
+  );
+
+  const chsyncKey = new gcp.serviceaccount.Key(
+    `${resourcePrefix}chsync-sa-key`,
+    {
+      serviceAccountId: chsyncSa?.accountId || '',
+    },
+  );
+
+  return {chsyncSa, chsyncKey};
+}
+
+function deployClickhouseSyncKubernetesResources(
+  name: string,
+  namespace: string | Input<string>,
+  propsString: Output<string>,
+  chsyncKey: gcp.serviceaccount.Key | undefined,
+  {
+    limits: requests = {
+      cpu: '1',
+      memory: '1024Mi',
+    },
+    env = [],
+    image,
+    resourcePrefix = '',
+    provider,
+    isAdhocEnv,
+    disableHealthCheck,
+  }: Pick<
+    OptionalArgs,
+    | 'limits'
+    | 'env'
+    | 'image'
+    | 'resourcePrefix'
+    | 'provider'
+    | 'isAdhocEnv'
+    | 'disableHealthCheck'
+  > = {},
+): void {
+  const propsHash = propsString.apply((props) =>
+    createHash('md5').update(props).digest('hex'),
+  );
+
+  const chsyncProps = new k8s.core.v1.Secret(
+    `${resourcePrefix}chsync-props`,
+    {
+      metadata: {
+        name: `${name}-chsync-props`,
+        namespace,
+      },
+      data: {
+        'config.yml': propsString.apply((str) =>
+          Buffer.from(str).toString('base64'),
+        ),
+      },
+    },
+    {provider},
+  );
+
+  const labels: Input<{
+    [key: string]: Input<string>;
+  }> = {
+    parent: name,
+    app: 'chsync',
+  };
+
+  const volumes: k8s.types.input.core.v1.Volume[] = [
+    {
+      name: 'props',
+      secret: {
+        secretName: chsyncProps.metadata.name,
+      },
+    },
+  ];
+  const volumeMounts: k8s.types.input.core.v1.VolumeMount[] = [
+    {name: 'props', mountPath: '/'},
+  ];
+
+  const initContainers: k8s.types.input.core.v1.Container[] = [];
+
+  // If service account is provided
+  if (chsyncKey) {
+    const chsyncSecretSa = new k8s.core.v1.Secret(
+      `${resourcePrefix}chsync-secret-sa`,
+      {
+        metadata: {
+          name: `${name}-chsync-sa`,
+          namespace,
+        },
+        data: {
+          'key.json': chsyncKey?.privateKey || '',
+        },
+      },
+      {provider},
+    );
+    volumes.push({
+      name: 'service-account-key',
+      secret: {
+        secretName: chsyncSecretSa.metadata.name,
+      },
+    });
+    volumeMounts.push({
```

Comment:
Why do we need google service key here?

Reply:
not sure. it was used in the debezium code. I will remove it 👍

Reply:
debezium connects to pubsub so it's needed